mypy/modulefinder.py

```diff
@@ -443,10 +444,23 @@ def find_modules_recursive(self, module: str) -> List[BuildSource]:
         names = sorted(self.fscache.listdir(package_path))
         for name in names:
             # Skip certain names altogether
-            if name == '__pycache__' or name.startswith('.') or name.endswith('~'):
+            if (
+                name in ("__pycache__", "site-packages", "node_modules")
+                or name.startswith(".")
+                or name.endswith("~")
+            ):
                 continue
             subpath = os.path.join(package_path, name)
 
+            if self.options and self.options.exclude:
+                subpath_str = os.path.abspath(subpath).replace(os.sep, "/")
+                if self.fscache.isdir(subpath):
+                    subpath_str += "/"
+                if re.search(self.options.exclude, subpath_str):
+                    if self.options.verbosity >= 2:
+                        print("TRACE: Excluding {}".format(subpath_str), file=sys.stderr)
+                    continue
```

Comment:
Can you share the functionality with `find_sources.py` by moving the code to a utility function?